In [1]:
import pandas as pd

Starting with subsets of the data 

In [5]:
beauty_dataset=pd.read_json('data/ref/All_Beauty_5.json', lines=True)
beauty_dataset.head(5)

overall  verified   reviewTime      reviewerID        asin  \
0        5      True   09 1, 2016  A3CIUOJXQ5VDQ2  B0000530HU   
1        5      True  11 14, 2013  A3H7T87S984REU  B0000530HU   
2        1      True  08 18, 2013  A3J034YH7UG4KT  B0000530HU   
3        5     False   05 3, 2011  A2UEO5XR3598GI  B0000530HU   
4        5      True   05 6, 2011  A3SFRT223XXWF7  B00006L9LC   

                                               style     reviewerName  \
0  {'Size:': ' 7.0 oz', 'Flavor:': ' Classic Ice ...         Shelly F   
1  {'Size:': ' 7.0 oz', 'Flavor:': ' Classic Ice ...     houserules18   
2  {'Size:': ' 7.0 oz', 'Flavor:': ' Classic Ice ...             Adam   
3  {'Size:': ' 7.0 oz', 'Flavor:': ' Classic Ice ...           Rich K   
4                          {'Size:': ' 200ml/6.7oz'}  C. C. Christian   

                                          reviewText  \
0                   As advertised. Reasonably priced   
1  Like the oder and the feel when I put it on my...   
2  I bought this to smell nice after I shave.  Wh...   
3  HEY!! I am an Aqua Velva Man and absolutely lo...   
4  If you ever want to feel pampered by a shampoo...   

                                             summary  unixReviewTime vote  \
0                                         Five Stars      1472688000  NaN   
1                                  Good for the face      1384387200  NaN   
2                                       Smells awful      1376784000  NaN   
3  Truth is There IS Nothing Like an AQUA VELVA MAN.      1304380800   25   
4                                    Bvlgari Shampoo      1304640000    3   

  image  
0   NaN  
1   NaN  
2   NaN  
3   NaN  
4   NaN

In [6]:
beauty_dataset['style']

0       {'Size:': ' 7.0 oz', 'Flavor:': ' Classic Ice ...
1       {'Size:': ' 7.0 oz', 'Flavor:': ' Classic Ice ...
2       {'Size:': ' 7.0 oz', 'Flavor:': ' Classic Ice ...
3       {'Size:': ' 7.0 oz', 'Flavor:': ' Classic Ice ...
4                               {'Size:': ' 200ml/6.7oz'}
                              ...                        
5264                       {'Design:': ' ETA-C6B5F7C374'}
5265                       {'Design:': ' ETA-7117EE3788'}
5266                       {'Design:': ' ETA-BF66BD2F87'}
5267                       {'Design:': ' ETA-03868FCB34'}
5268                       {'Design:': ' ETA-C4893097E9'}
Name: style, Length: 5269, dtype: object

In [7]:
beauty_dataset.drop(['style','image'], axis=1).describe(include='all') #style and image are causing issues with describe()

overall verified   reviewTime     reviewerID        asin  \
count   5269.000000     5269         5269           5269        5269   
unique          NaN        2          883            991          85   
top             NaN     True  07 14, 2016  AKJHHD5VEH7VG  B0012Y0ZG2   
freq            NaN     4280           33             12        1794   
mean       4.771873      NaN          NaN            NaN         NaN   
std        0.743204      NaN          NaN            NaN         NaN   
min        1.000000      NaN          NaN            NaN         NaN   
25%        5.000000      NaN          NaN            NaN         NaN   
50%        5.000000      NaN          NaN            NaN         NaN   
75%        5.000000      NaN          NaN            NaN         NaN   
max        5.000000      NaN          NaN            NaN         NaN   

           reviewerName     reviewText     summary  unixReviewTime vote  
count              5269           5264        5264    5.269000e+03  552  
unique              905           1363        1011             NaN   24  
top     Amazon Customer  Great product  Five Stars             NaN    2  
freq                362             15        1455             NaN  238  
mean                NaN            NaN         NaN    1.441506e+09  NaN  
std                 NaN            NaN         NaN    5.473127e+07  NaN  
min                 NaN            NaN         NaN    1.095725e+09  NaN  
25%                 NaN            NaN         NaN    1.407024e+09  NaN  
50%                 NaN            NaN         NaN    1.445645e+09  NaN  
75%                 NaN            NaN         NaN    1.481674e+09  NaN  
max                 NaN            NaN         NaN    1.534982e+09  NaN

The dataset contains 5264 reviews for 85 Beauty products from 991 reviewers

In [14]:
iphone_dataset=pd.read_json('data/apple_iphone_11_reviews.json', orient='records')
iphone_dataset.head()

product                    helpful_count  \
0  Apple iPhone XR (64GB) - Black  5,087 people found this helpful   
1  Apple iPhone XR (64GB) - Black  2,822 people found this helpful   
2  Apple iPhone XR (64GB) - Black  1,798 people found this helpful   
3  Apple iPhone XR (64GB) - Black  1,366 people found this helpful   
4  Apple iPhone XR (64GB) - Black    536 people found this helpful   

   total_comments                                                url  \
0              24  https://www.amazon.in/Apple-iPhone-XR-64GB-Bla...   
1               6  https://www.amazon.in/Apple-iPhone-XR-64GB-Bla...   
2               0  https://www.amazon.in/Apple-iPhone-XR-64GB-Bla...   
3              14  https://www.amazon.in/Apple-iPhone-XR-64GB-Bla...   
4               5  https://www.amazon.in/Apple-iPhone-XR-64GB-Bla...   

  review_country reviewed_at  \
0         India   2018-12-12   
1         India   2018-11-17   
2         India   2019-01-27   
3         India   2019-05-02   
4         India   2019-05-24   

                                         review_text       review_rating  \
0                                              NOTE:  3.0 out of 5 stars   
1  Very bad experience with this iPhone xr phone....  1.0 out of 5 stars   
2  Amazing phone with amazing camera coming from ...  5.0 out of 5 stars   
3  So I got the iPhone XR just today. The product...  1.0 out of 5 stars   
4  I've been an android user all my life until I ...  5.0 out of 5 stars   

  product_company     profile_name  \
0           Apple     Sameer Patil   
1           Apple  Amazon Customer   
2           Apple                A   
3           Apple    Shubham Dutta   
4           Apple     Nepuni Lokho   

                                        review_title  
0  Which iPhone you should Purchase ? iPhone 8, X...  
1                   Don't buy iPhone xr from Amazon.  
2                            Happy with the purchase  
3  Amazon is not an apple authorised reseller. Pl...  
4       Excellent Battery life and buttery smooth UI

In [16]:
iphone_dataset.describe(include='all')

product helpful_count  total_comments  \
count                             5010          5010     5010.000000   
unique                               1            49             NaN   
top     Apple iPhone XR (64GB) - Black             0             NaN   
freq                              5010          4270             NaN   
first                              NaN           NaN             NaN   
last                               NaN           NaN             NaN   
mean                               NaN           NaN        0.030938   
std                                NaN           NaN        0.589420   
min                                NaN           NaN        0.000000   
25%                                NaN           NaN        0.000000   
50%                                NaN           NaN        0.000000   
75%                                NaN           NaN        0.000000   
max                                NaN           NaN       24.000000   

                                                      url review_country  \
count                                                5010           5010   
unique                                                500              1   
top     https://www.amazon.in/Apple-iPhone-XR-64GB-Bla...         India    
freq                                                   20           5010   
first                                                 NaN            NaN   
last                                                  NaN            NaN   
mean                                                  NaN            NaN   
std                                                   NaN            NaN   
min                                                   NaN            NaN   
25%                                                   NaN            NaN   
50%                                                   NaN            NaN   
75%                                                   NaN            NaN   
max                                                   NaN            NaN   

                reviewed_at review_text       review_rating product_company  \
count                  5010        5010                5010            5010   
unique                  411        4089                   5               1   
top     2019-11-03 00:00:00        Good  5.0 out of 5 stars           Apple   
freq                    107         146                3731            5010   
first   2018-11-01 00:00:00         NaN                 NaN             NaN   
last    2020-04-22 00:00:00         NaN                 NaN             NaN   
mean                    NaN         NaN                 NaN             NaN   
std                     NaN         NaN                 NaN             NaN   
min                     NaN         NaN                 NaN             NaN   
25%                     NaN         NaN                 NaN             NaN   
50%                     NaN         NaN                 NaN             NaN   
75%                     NaN         NaN                 NaN             NaN   
max                     NaN         NaN                 NaN             NaN   

           profile_name review_title  
count              5010         5010  
unique             4099         3104  
top     Amazon Customer         Good  
freq                550          209  
first               NaN          NaN  
last                NaN          NaN  
mean                NaN          NaN  
std                 NaN          NaN  
min                 NaN          NaN  
25%                 NaN          NaN  
50%                 NaN          NaN  
75%                 NaN          NaN  
max                 NaN          NaN

In this dataset, we have 5010 reviews of Apple Iphone 11. 
A couple of comments :
* The reviews are coming from a very large number of different customers
* The structure is very similar to the beauty products dataset even though the source is not exactly the same
* We can drop the product_company column, it does not bring any value

In [19]:
appliances_dataset=pd.read_json('data/ref/Appliances_5.json', lines=True)
appliances_dataset.head()

overall  verified   reviewTime      reviewerID        asin  \
0        5      True  08 22, 2013  A34A1UP40713F8  B00009W3I4   
1        5      True   02 8, 2016  A1AHW6I678O6F2  B00009W3PA   
2        5      True   08 5, 2015   A8R48NKTGCJDQ  B00009W3PA   
3        5      True  04 24, 2015   AR3OHHHW01A8E  B00009W3PA   
4        5      True  03 21, 2015  A2CIEGHZ7L1WWR  B00009W3PA   

                       style     reviewerName  \
0  {'Style:': ' Dryer Vent'}    James. Backus   
1       {'Size:': ' 6-Foot'}           kevin.   
2       {'Size:': ' 6-Foot'}        CDBrannom   
3       {'Size:': ' 6-Foot'}  Calvin E Reames   
4       {'Size:': ' 6-Foot'}   albert j. kong   

                                          reviewText        summary  \
0  I like this as a vent as well as something tha...  Great product   
1                                          good item     Five Stars   
2                     Fit my new LG dryer perfectly.     Five Stars   
3                     Good value for electric dryers   Perfect size   
4                  Price and delivery was excellent.     Five Stars   

   unixReviewTime vote image  
0      1377129600  NaN   NaN  
1      1454889600  NaN   NaN  
2      1438732800  NaN   NaN  
3      1429833600  NaN   NaN  
4      1426896000  NaN   NaN

Same source than beauty dataset 

The beauty and the appliances datasets are subsets of a bigger dataset. For this main dataset, metada are provided to give more info about the products. Let's explore those metada

In [22]:
meta_data=pd.read_json('data/ref/meta_data.json', lines=True)
meta_data.head()

description  \
0      [Brand new. Still in factory-sealed plastic.]   
1  [If you're already a Photoshop user and want a...   
2  [School To Home Notes: Note From the Teacher!\...   
3  [Dooqi is a young brand specializing in high-e...   
4                                [Cover for Nook HD]   

                                               title  \
0  Kelby Training DVD: The Adobe Photoshop CS5 7-...   
1  Kelby Training DVD: Adobe Photoshop CS5 Power ...   
2  TEACHERS FRIEND NOTE FROM THE TEACHER SCHOOL T...   
3  [3-pack]Ipad Mini Screen Protector, Dooqi Scre...   
4                            Hd+ Cheetah Cover- Nook   

                                               image            brand  \
0  [https://images-na.ssl-images-amazon.com/image...   Kelby Training   
1  [https://images-na.ssl-images-amazon.com/image...   Kelby Training   
2  [https://images-na.ssl-images-amazon.com/image...  TEACHERS FRIEND   
3  [https://images-na.ssl-images-amazon.com/image...            dooqi   
4                                                NaN             Nook   

                                                rank   main_cat       date  \
0  [>#654,681 in Electronics (See Top 100 in Elec...  Computers 2011-12-02   
1  [>#830,165 in Electronics (See Top 100 in Elec...  Computers 2011-11-07   
2  [>#1,233,887 in Electronics (See Top 100 in El...  Computers 2007-06-06   
3  [>#454,595 in Electronics (See Top 100 in Elec...  Computers 2014-12-05   
4  [>#85,661 in Computers & Accessories (See top ...  Computers 2013-09-11   

         asin                                            feature  \
0  0321732960                                                NaN   
1  0321735722                                                NaN   
2  0439651859                                                NaN   
3  0545729971  [Hard coated for Scratch Proof (Hardness: 4H),...   
4  0594450284                                    [Cheetah cover]   

                                               tech1 also_buy price also_view  \
0                                                NaN      NaN   NaN       NaN   
1                                                NaN      NaN   NaN       NaN   
2                                                NaN      NaN   NaN       NaN   
3                                                NaN      NaN   NaN       NaN   
4  {'Brand Name': 'Nook', 'Item Weight': '1 pound...      NaN   NaN       NaN   

  tech2 details similar_item  
0   NaN     NaN          NaN  
1   NaN     NaN          NaN  
2   NaN     NaN          NaN  
3   NaN     NaN          NaN  
4   NaN     NaN          NaN

In [25]:
meta_data[['asin','description','title','date','brand']].describe(include='all')

TypeError: unhashable type: 'list'

Exception ignored in: 'pandas._libs.index.IndexEngine._call_map_locations'
Traceback (most recent call last):
  File "pandas\_libs\hashtable_class_helper.pxi", line 1652, in pandas._libs.hashtable.PyObjectHashTable.map_locations
TypeError: unhashable type: 'list'


asin description                         title  \
count        18772        5130                         18771   
unique       18521        4437                         18417   
top     B00006L39I          []  onepix Hard Case for MacBook   
freq             2         122                             7   
first          NaN         NaN                           NaN   
last           NaN         NaN                           NaN   

                       date  brand  
count                 18588  18295  
unique                 3042   5785  
top     2016-06-16 00:00:00     HP  
freq                    404    212  
first   1973-09-04 00:00:00    NaN  
last    2019-01-08 00:00:00    NaN

There are a couple of duplicates in the dataset

For more info about the dataset, please check https://nijianmo.github.io/amazon/index.html